lib/livebook_web/live/path_select_component.ex

```diff
@@ -97,19 +127,21 @@ defmodule LivebookWeb.PathSelectComponent do
                 <%= remix_icon("folder-add-fill", class: "text-xl align-middle text-gray-400") %>
               </span>
               <span class="flex font-medium text-gray-500">
-                <form phx-submit="create_directory" phx-target="<%= @myself %>">
+                <div
+                  phx-window-keydown="cancel_new_directory"
+                  phx-key="escape"
+                  phx-target="<%= @myself %>">
```

Comment:
Note: apparently when submitting the form LV blurs then focuses the input, which triggers `phx-blur` (cancellation in our case). I will submit a corresponding issue, but since we only use a single output I managed to use Enter keydown event instead.